chore: Enable StandaloneWindows64 tests for Unity.Netcode.RuntimeTests testfilter

zain-mecklai opened the pull request:
Building and running the standalone players is exposing some interesting issues with the way tests are set up so I'm  making this part of the nightly run so we can catch any issues that might be introduced.

zain-mecklai committed:
Enable StandaloneWindows64 tests for Unity.Netcode.RuntimeTests testfilter

zain-mecklai committed:
chore: Enable StandaloneWindows64 tests for Unity.Netcode.RuntimeTests testfilter

zain-mecklai committed:
Rename the directories to match platform.name

zain-mecklai committed:
Give the standalone tests a unique name

zain-mecklai committed:
Make sure to run setup before running build or test

zain-mecklai committed:
Support setup as an argument

zain-mecklai committed:
Remove windows as it is replaced by win

zain-mecklai committed:
Support for ubuntu standalone

zain-mecklai committed:
don't need Android

zain-mecklai committed:
Merge branch 'chore/enable-standalone-player-tests' of github.com:Unity-Technologies/com.unity.netcode.gameobjects into chore/enable-standalone-player-tests

zain-mecklai committed:
Merge branch 'develop' into chore/enable-standalone-player-tests

zain-mecklai committed:
platform specific path separators

zain-mecklai committed:
Replace tabs with spaces

zain-mecklai committed:
Replace tabs with spaces and equalize spaces

zain-mecklai committed:
Fix formatting, again

zain-mecklai committed:
Rename job

zain-mecklai committed:
Apparently I don't know the difference between windows and linux

zain-mecklai committed:
pip install the unity downloader, assuming pip is readily available

zain-mecklai committed:
add mac and make utr executable

zain-mecklai committed:
Merge branch 'chore/enable-standalone-player-tests' of github.com:Unity-Technologies/com.unity.netcode.gameobjects into chore/enable-standalone-player-tests

zain-mecklai committed:
Run without graphics in batchmode

zain-mecklai committed:
use a graphics enabled machine for this test

zain-mecklai committed:
Export the display to :0 for Ubuntu

JesseOlmer commented on .yamato/standalone-project-tests.yml:
```diff
@@ -0,0 +1,35 @@
+{% for project in projects -%}
+{% for editor in project.test_editors -%}
+{% for platform in test_platforms -%}
+standalone_test_{{ project.name }}_{{ editor }}_{{ platform.name }}:
```
This feels like the wrong approach. You're doing a bunch of work to call into a shell script that ultimately only runs a few commands, which are almost identical across platforms. This obfuscates some of what's going on, it duplicates a bunch of code, and it introduces additional places where we have to make changes anytime we update things like editors or platforms. You also can't benefit from dependency graph stuff in the yamato job scheduler.

JesseOlmer replied:
I think if you, instead, had a build & test job just like you did with mobile you could do this all without the batch files.

The build & test commands are almost identical - the test one could look like this for example (build would be very similar):

```
{% for project in projects -%}
{% for editor in project.test_editors -%}
{% for platform in test_platforms -%}
standalone_test_{{ project.name }}_{{ editor }}_{{ platform.name }}:
  name : standalone {{ project.name }} tests - {{ editor }} on {{ platform.name }}
  agent:
    type: {{ platform.type }}
    image: {{ platform.image }}
    flavor: {{ platform.flavor}}
  commands:
    - {% if platform.name == "win" %}SET{% else %}EXPORT{% endif %} UTR_VERSION=0.12.0
    - {% if platform.name != "win" %}./{% endif %}utr --suite=playmode --platform={{ platform.utr_name }}  --editor-location=.Editor --testproject=testproject --player-save-path=build/players --artifacts_path=build/logs --scripting-backend=mono --build-only --testfilter=Unity.Netcode.RuntimeTests --extra-editor-arg=-batchmode --extra-editor-arg=-nographics
{% endfor -%}
{% endfor -%}
{% endfor -%}
```

zain-mecklai replied:
@JesseOlmer are you satisfied with the changes I've made here?

JesseOlmer commented on build-scripts/builder/mac/target/mac:
```diff
@@ -0,0 +1,24 @@
+        echo "setup"
+        curl -s https://artifactory.prd.it.unity3d.com/artifactory/unity-tools-local/utr-standalone/utr --output utr
+        chmod +x ./utr
+        unity-downloader-cli -u 2021.2 -c Editor --fast --wait
```
In the yamato job definition you're looping on test_editors but your shell scripts are all hard-coded to 2021.2. This will result in jobs for, e.g. trunk that download and run 2021.2 instead

zain-mecklai replied:
fixed. Now with the job inline, it uses the template variables the same way as everywhere else.

JesseOlmer commented on build-scripts/builder/ubuntu/target/ubuntu:
```diff
@@ -0,0 +1,23 @@
+    "build")
+        echo "build"
+        export UTR_VERSION=0.12.0
+        ./utr --suite=playmode --platform=StandaloneLinux64  --editor-location=.Editor --testproject=testproject --player-save-path=build/players --artifacts_path=build/logs --scripting-backend=mono --build-only --testfilter=Unity.Netcode.RuntimeTests --extra-editor-arg=-batchmode --extra-editor-arg=-nographics
```
You should be able to use `--extra-editor-arg=-batchmode --extra-editor-arg=-nographics` on all platforms and get a perf boost.

zain-mecklai replied:
added the extra editor args

JesseOlmer commented on .yamato/standalone-project-tests.yml:
```diff
@@ -0,0 +1,35 @@
+standalone_test_{{ project.name }}_{{ editor }}_{{ platform.name }}:
+  name : standalone {{ project.name }} tests - {{ editor }} on {{ platform.name }}
+  agent:
+    type: {{ platform.type }}{% if platform.name == "ubuntu" %}::GPU{% endif %}
```
Why are you asking for a GPU instance with an rtx2080 if the actual test command uses `-batchmode -nographics`?

zain-mecklai replied:
The build doesn't need it, but the run time crashes without a real graphics context.

zain-mecklai replied:
Should I separate build and test into different jobs?

zain-mecklai replied:
Ah, @JesseOlmer I finally have a better answer. The issue is that I'm launching the standalone player and the extra-editor-args don't get passed to that so the player still comes up and attempts to launch the ui which it can't. I'm asking around as to how to get the player to take args through utr but I don't know what else to do at the moment.

zain-mecklai commented:
@JesseOlmer I'll revert these changes and go back to using the inline commands in the job definition files instead of calling out to shell scripts. I'll get those changes in later today.

zain-mecklai committed:
inline all the build and test commands

zain-mecklai committed:
Delete scripts

zain-mecklai committed:
Clean up after review feedback

zain-mecklai committed:
Merge branch 'develop' into chore/enable-standalone-player-tests

zain-mecklai committed:
utr needs execute permissions on non-windows machines

zain-mecklai committed:
Need a linux machine with graphics since the player cannot currently be prevented from starting in graphics mode

zain-mecklai committed:
Merge branch 'develop' into chore/enable-standalone-player-tests

zain-mecklai committed:
Merge branch 'develop' into chore/enable-standalone-player-tests

zain-mecklai committed:
merge with develop

zain-mecklai committed:
remove requirement for a graphics card for testing

zain-mecklai committed:
try to use Null as the graphics api for the player

zain-mecklai committed:
Use graphics enabled Ubuntu machines for standalone tests

zain-mecklai committed:
Merge branch 'develop' into chore/enable-standalone-player-tests

zain-mecklai committed:
Merge branch 'develop' into chore/enable-standalone-player-tests

ShadauxCat commented:
@zain-mecklai Is this PR still relevant?

zain-mecklai commented:
> @zain-mecklai Is this PR still relevant?

Yes, I'm just not sure what I need to do to get it merged.

zain-mecklai committed:
Merge branch 'develop' into chore/enable-standalone-player-tests

zain-mecklai committed:
Merge branch 'develop' into chore/enable-standalone-player-tests

zain-mecklai committed:
Merge branch 'develop' into chore/enable-standalone-player-tests

zain-mecklai committed:
Remove the duplicate flavor field and use newer b1 values instead of m1

zain-mecklai committed:
Merge branch 'develop' into chore/enable-standalone-player-tests

zain-mecklai committed:
Merge branch 'develop' into chore/enable-standalone-player-tests

zain-mecklai committed:
Not sure why whitespace is flagged on the last line of file, try to clean up

zain-mecklai committed:
Merge branch 'develop' into chore/enable-standalone-player-tests

NoelStephensUnity committed:
fix

First pass attempt at fixing the issues with consoles failing on NetworkBehaviourUpdaterTests.

NoelStephensUnity committed:
update

removing some commented out code.

NoelStephensUnity committed:
style

fixing white spaces.

NoelStephensUnity committed:
style

minor update to XML Doc comments, removed param that no longer exists in BehaviourUpdaterAllTests

zain-mecklai committed:
Merge branch 'fix/NetworkBehaviourUpdaterTests-ConsoleFailures' into chore/enable-standalone-player-tests

zain-mecklai commented:
merged in changes for Noel for failures in this test for consoles to see if that helps.

NoelStephensUnity committed:
test update

Making some changes to see if mac sometimes still had remaining GameObjects hanging around when the next test started...added additional client side info debug output in the event a timeout happens.

zain-mecklai committed:
Archive the artifacts path

NoelStephensUnity committed:
test update

Waiting 1 tick period to assure spawned objects are spawned on standalone build tests.

NoelStephensUnity committed:
test refactor

This is a complete refactoring of the NetworkBehaviourUpdaterTests.

NoelStephensUnity committed:
test fix

This should run fine on a stand alone test runner build.

NoelStephensUnity reviewed (approved):
Looks good Zain!

NoelStephensUnity committed:
Merge branch 'develop' into chore/enable-standalone-player-tests

NoelStephensUnity committed:
merge fix

Minor fix for a bad merge.

NoelStephensUnity committed:
Merge branch 'develop' into chore/enable-standalone-player-tests

zain-mecklai commented:
@JesseOlmer any chance you can take a look at this and decide if it is ready to go or if it needs more changes?

zain-mecklai commented:
@lpmaurice can you help me shepherd this change into develop branch please?

zain-mecklai committed:
Merge branch 'develop' into chore/enable-standalone-player-tests
